fluent/syntax/serializer.py

```diff
@@ -0,0 +1,249 @@
+
+def serialize_value(pattern, indent=0):
+    multi = bool(filter(
+        contains_new_line, pattern.elements
+    ))
+
+    schema = " {}"
+
+    if (multi):
+        indent += 1
+        schema = "\n{}| {{}}".format(" " * 4 * indent)
+
+    return schema.format(
+        serialize_pattern(pattern, indent, multi)
+    )
+
+
+def serialize_pattern(pattern, indent=0, multi=False):
+    schema = "\"{}\"" if pattern.quoted else "{}"
+    return schema.format(
+        "".join([
+            serialize_element(elem, indent, multi)
+            for elem in pattern.elements
+        ])
+    )
+
+
+def serialize_element(element, indent=0, multi=False):
+    if isinstance(element, ast.StringExpression):
+        return serialize_string_expression(element, indent, multi)
+    if isinstance(element, ast.Pattern):
+        return "{{ {} }}".format(
+            serialize_pattern(element, indent)
+        )
+    if isinstance(element, ast.SelectExpression):
+        return "{{{}}}".format(
+            serialize_select_expression(element, indent)
+        )
+    if isinstance(element, ast.Expression):
+        return "{{ {} }}".format(
+            serialize_expression(element)
+        )
+
+
+def serialize_expression(expression):
+    if isinstance(expression, ast.Pattern):
+        return serialize_pattern(expression)
+    if isinstance(expression, ast.NumberExpression):
+        return serialize_number_expression(expression)
+    if isinstance(expression, ast.MessageReference):
+        return serialize_message_reference(expression)
+    if isinstance(expression, ast.ExternalArgument):
+        return serialize_external_argument(expression)
+    if isinstance(expression, ast.AttributeExpression):
+        return serialize_attribute_expression(expression)
+    if isinstance(expression, ast.VariantExpression):
+        return serialize_variant_expression(expression)
+    if isinstance(expression, ast.CallExpression):
+        return serialize_call_expression(expression)
+
+
+def serialize_string_expression(expr, indent=0, multi=False):
+    if multi:
+        return "{}| ".format(" " * 4 * indent).join(
+            expr.value.splitlines(True)
+        )
+    else:
+        return expr.value
+
+
+def serialize_number_expression(expr):
+    return expr.value
+
+
+def serialize_message_reference(expr):
+    return serialize_identifier(expr.id)
+
+
+def serialize_external_argument(expr):
+    return "${}".format(serialize_identifier(expr.id))
+
+
+def serialize_select_expression(expr, indent=0):
+    parts = []
+
+    if expr.expression:
+        selector = " {} ->".format(
+            serialize_expression(expr.expression)
+        )
+        parts.append(selector)
+
+    for variant in expr.variants:
+        parts.append(serialize_variant(variant, indent + 1))
+
+    parts.append("\n{}".format(" " * 4 * indent))
+
+    return "".join(parts)
+
+
+def serialize_variant(variant, indent=0):
+    return "\n{}{}[{}]{}".format(
+        " " * (4 * indent - 1),
```

Comment:
oh, sorry, I mean, I'd prefer that non-default variants be `"    " * indent`. and default `*` be `"    " * (indent - 1) + "   *"`.